dashboard/src/components/DeploymentForm/BasicDeploymentForm/BasicDeploymentForm.tsx

```diff
@@ -27,99 +28,58 @@ class BasicDeploymentForm extends React.Component<IBasicDeploymentFormProps> {
-          />
-        );
-      case "password":
-        return (
-          <TextParam
-            label="Password"
-            handleBasicFormParamChange={this.props.handleBasicFormParamChange}
-            key={id}
-            id={id}
-            name={name}
-            param={param}
-          />
-        );
-      case "email":
-        return (
-          <TextParam
-            label="Email"
-            handleBasicFormParamChange={this.props.handleBasicFormParamChange}
-            key={id}
-            id={id}
-            name={name}
-            param={param}
-          />
-        );
-      case "externalDatabase":
-        return (
-          <DatabaseSection
-            label="External Database Details"
+          <Subsection
+            label={param.title || "External Database Details"}
             handleValuesChange={this.props.handleValuesChange}
             appValues={this.props.appValues}
             key={id}
+            name={name}
             param={param}
+            enablerChildrenParam={USE_SELF_HOSTED_DB}
+            enablerCondition={false}
           />
         );
-      case "diskSize":
+      case DISK_SIZE:
         return (
-          <DiskSizeParam
-            label="Disk Size"
+          <SliderParam
+            label={param.title || "Disk Size"}
             handleBasicFormParamChange={this.props.handleBasicFormParamChange}
             key={id}
             id={id}
             name={name}
             param={param}
+            min={1}
+            max={100}
+            unit="Gi"
           />
         );
       default:
         switch (param.type) {
-          case "string":
+          case "boolean":
             return (
-              <TextParam
-                label={param.title || ""}
+              <BooleanParam
+                label={param.title || name}
                 handleBasicFormParamChange={this.props.handleBasicFormParamChange}
                 key={id}
                 id={id}
                 name={name}
                 param={param}
               />
             );
-          case "integer":
+          default:
             return (
               <TextParam
-                label={param.title || ""}
-                handleBasicFormParamChange={this.props.handleBasicFormParamChange}
-                key={id}
-                id={id}
-                name={name}
-                param={param}
-                inputType="number"
-              />
-            );
-          case "boolean":
-            return (
-              <BooleanParam
-                label={param.title || ""}
+                label={param.title || name}
                 handleBasicFormParamChange={this.props.handleBasicFormParamChange}
                 key={id}
                 id={id}
                 name={name}
                 param={param}
+                inputType={param.type === "integer" ? "number" : "string"}
               />
             );
-          default:
```

Comment:
I have removed this case to assume that if the `type` is not set, it refers to a `string`